docs/Practices/hub/hub.md

```diff
@@ -0,0 +1,118 @@
+
+- [UA: Clean Architecture](./learning.md#cleanhexagonalonion-architecture), [UA: Repository pattern](./learning.md#repository-pattern), [UA: Persistence Agnosticism](./learning.md#persistence-agnosticism) Repository layer code is the only code that should depend on types representing the shape of database table/view/query/etc. *The implementation details of data persistence should not be exposed to consumers of a domain model that utilizes said persistence. One of the quickest ways to a building a ball of mud is to do let a database schema (slow and risky to change) be the source of truth for the shape of a domain model across all layers and systems. There are cases where this is ok (constrained problem space, prototypes, etc), but this should be avoided by default.*
+
+- [UA: Slim APIs are Easier to Maintain](./learning.md#slim-apis-are-easier-to-maintain) Don't require more data for a function than is required. *A slim API is less risky to change and easier maintain and reason about.*
+
+- [UA: Domain Driven Design - Bounded Contexts](./learning.md#bounded-contexts), [UA: Eventual Consistency](./learning.md#eventual-consistency) All data should only have a single source of truth. *It is ok for data to be stored in multiple places as long as the source of truth is always one location. For concepts that exist within multiple domains (students), it is perfectly acceptable for that concept to have it's source of truth spread across data stores as long as no part (field) of any individual meaning is has it's source of truth in more than one location.*
+
+- [UA: Errors as Values](./learning.md#errors-as-values), [UA: Domain Driven Design - Domain Modeling](./learning.md#domain-modeling) The known errors of a command/query within a Vertical Slice should be included in its type signature. *It is recommended to coerce exceptions to values for known errors whenever possible, but at minimum no known errors should be exposed from a command/query function by way of thrown exception.*
+
+- [UA: Domain Driven Design - Domain Modeling](./learning.md#domain-modeling) If an error represents a possible bad data state there should be documentation regarding this bad data state and how to remediate it. *At minimum there should be a comment attached to the type definition for a given error that provides appropriate context.*
+
+- [UA: Testing](./learning.md#testing) When a bug is fixed an existing test should be updated to "prove" the fix or a new test should be written to accompany the fix.
+
+- [UA: Validation](./learning.md#validatation) Validation should be performed on all handler request data.
```

Comment:
Meaning we get the data we expect string/int/etc? Or that we are validating the data against the DB?

Reply:
Would you consider publishing the "Validate for:" section in the doc itself? I found your response to Skylar's question very useful.

Reply:
Added